### ISSUE TITLE
mail_production.log file not being created; mail errors not being logged

### ISSUE THREAD

**Issue by dgwmfo** [joshmcarthur/mail-logger#2]
Hi Josh:

I'm struggling with a recurrent EOFError when I try to send emails from my Rails app via our corporate smtp server. It works in the production version of the app on an Ubuntu 12.04 server in our building, but in the remote building where I just installed another server running a similar version of the app, it doesn't work.

To try to get a better feel for what's going wrong, I added your mail-logger gem to the app's Gemfile and deployed to the remote production server. When I generate the error there (I post an update on a project and a notification email is supposed to go out) I'm not seeing the log file  I'd expected your app to create in RAILS_ROOT/current/log.

I'm running rails 4.0.4, ruby 2.0.0 (under rbenv) and Ubuntu 12.04 LTS.

Any help would be greatly appreciated... your app looks like it fulfills an important need!

Thanks!

--Dean Richardson

**Comment by joshmcarthur** [comment]
Hi Dean,

Sorry to hear you're having problems!
The first thing that I'd like you to check is that you've got the most recent version of the gem - `0.0.4`, as that release fixed an issue where the logs were not written to until the process generating the log file content terminated (for example, `rails server`).

The next thing that would be worth checking is to check that the original error you are getting is definitely in the mailing part of your application, and not earlier in the code. The mail-logger gem works by hooking into the [mail gem](http://rubygems.org/gems/mail) to receive a callback _after_ the mail is delivered - so if your problem is occurring before the mail sends, then it won't receive a copy of the mail to log.  I'm just wondering whether a filesystem or encoding problem could be occurring some where in your code (the mailer or mailer templates perhaps?), that might cause the mailer to error with the `EOFError` you've been seeing before it actually sends.

Finally, for debugging purposes, you might want to try sending your email directly from a Rails console on the remote server, if you have access to that - that might give you a little more information about the origin of the error.

**Comment by dgwmfo** [comment]
Hi Josh:

I appreciate the thorough reply. I did end up trying to email from the console, and that ultimately helped a little. But eventually, because the apps I develop for internal use have to hook into our corporate email relay server, I tried telneting to it over port 25. After some trial and error, that uncovered the real problem, which ended up just being an authentication error with the relay server. Now that’s fixed and everything’s working great.

Thanks,

Dean
